### PR TITLE
Fix test keybinding for layer actions

### LINKED ISSUE #5404: problematic side-effects in test functions
## 🐛 Bug

While working on #4894, I noticed [some odd failures in tests](https://github.com/napari/napari/pull/4894#issuecomment-1346620948) which behaved differently depending on which tests were being run. Turns out, the issue is on `main` too, it was just changed slightly in that PR which exposed the issue. Get ready for some weird stuff :)

If you run `pytest napari/_tests/` (or the full test suite) on `main`, everything looks fine. However,

```bash
pytest napari/_tests/ -k "test_expected_number_of_layer_methods"
```

will fail on image and shapes. Later, I found out that this:

```bash
pytest napari/_tests/ -k "test_expected_number_of_layer_methods or test_mgui_forward_refs"
```

will instead fail on points, labels and shapes, and succeed on the rest. If you run any other function in the same file, however, we keep the original failure:

```bash
pytest napari/_tests/ -k "test_expected_number_of_layer_methods or test_magicgui_get_viewer"
```

This suggests that `test_mgui_forward_refs` has some side effects which change the behaviour of subsequent tests. However, whatever it does, it's not enough to get to the "all tests pass" behaviour we get from the full test suite. So I locked in the above tests, and looked for another function with side effects.

Long story short, the functions inside `test_mouse_bindings.py` need to run *before* the magicgui one. So if you run:

```bash
pytest napari/_tests/ -k "test_expected_number_of_layer_methods or test_mgui_forward_ref or _mouse_bindings"
```

It will fail, but if you run:

```bash
pytest napari/_tests/{test_mouse_bindings.py,test_magicgui.py,test_viewer.py}
```

You finally get the "successful" behaviour. Note that between the two last commands, only the order of execuction changed.

---

Any suggestions on how to fix this? It seems likely that monkeypatches and `Mock` are misbehaving.

cc @andy-sweet @Czaki

### PR DESCRIPTION
**Pull request by brisvag**
# Description
Fixes #5404.

I changed the test so they actually do what they were intended to: ensuring that keybinding existed for each action (rather than having hardcoded numbers like we have now).

To do so I had to programmatically find all relevant actions and shortcuts.

This fixes a few things:
- the result helped me track down a couple of missing shortcuts (which the test was meant to catch) and a quasi-duplicated shortcut for `Image.plane`.
- I found out what the issue was with #5404: layers need to be first instantiated once in order to be correctly populated. Now the test is deterministic. Not sure if we need to somehow fix the behaviour so that instantiation is not required, but that's out of scope for this PR.
- tests are hopefully clearer in intent now that I renamed and restructed them a bit

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: napari/_tests/test_viewer.py [hunk 1]
```diff
@@ -95,26 +103,28 @@ def test_add_layer(make_napari_viewer, layer_class, data, ndim):
...
+def test_all_layer_actions_are_accessible_via_shortcut(
+    layer_class, data, ndim
+):
     """
-    Make sure we do find all the methods attached to a layer via keybindings
+    Make sure we do find all the actions attached to a layer via keybindings
     """
-    layer_methods = _get_all_keybinding_methods(getattr(layers, cls))
-    assert len(layer_methods) == expectation
+    _ = layer_class(
```
Comment (Czaki) [inline napari/_tests/test_viewer.py]: Why assignment to `_`?

File: napari/_tests/test_viewer.py [hunk 2]
```diff
@@ -12,33 +12,41 @@
...
-    _ = make_napari_viewer()
-    viewer_methods = _get_all_keybinding_methods(Viewer)
-    assert len(viewer_methods) == EXPECTED_NUMBER_OF_VIEWER_METHODS
+    _ = (
+        make_napari_viewer()
+    )  # instantiate to make sure everything is initialized correctly
+    _assert_shortcuts_exist_for_each_action(Viewer)
 
 
 @pytest.mark.xfail
```
Comment (Czaki) [inline napari/_tests/test_viewer.py:56]: is this xfail required?

**Review by Czaki — APPROVED** [review]
I like this change. Some plugins will not fail fixed tests.

File: napari/_tests/test_viewer.py (hunk 2 above)
Comment (brisvag) [reply napari/_tests/test_viewer.py:56]: Not sure, I left this untouched from before. I think the this boils down to the same issue below where you need to first instantiate the object for actions to be correctly initialized. I think we can ust remove it?

File: napari/_tests/test_viewer.py (hunk 1 above)
Comment (brisvag) [reply napari/_tests/test_viewer.py]: we just need to initialize it, but it's otherwise useless.

**Comment by brisvag** [comment]
Thanks to the codecov warnings I noticed that it wasn't being run correctly. Now it's fixed! We needed to navigate the full `mro()` to get any actions.

File: napari/_tests/test_viewer.py (hunk 2 above)
Comment (brisvag) [reply napari/_tests/test_viewer.py:56]: Actually, probably better to leave it, it shows that we have some issue with intitialization so we don't forget to solve it at some point. I'd say solving it is out of scope for this PR, though.

**Comment by andy-sweet** [comment]
Sorry, I didn't have time to review this, but I am so happy those magic method counts are gone now!

**Pull request merged**